### DISCUSSION TITLE
Update Query after mutation requires variables

### DISCUSSION THREAD

**Discussion by edw19** [urql-graphql/urql#2011] [Q&A] [answered]
I have this code in my cache

![image](https://user-images.githubusercontent.com/47750859/136099413-8cd61de6-8bcb-4364-8ac6-fdc8de4e198e.png)

the problem is that when I want to update the GET_SALES_AND_TOTALS query it requires variables that I cannot get in the cache (queryBy, startDate, endDate)
am i misusing the update query method or is there a better solution

**Comment by JoviDeCroock** [comment, marked as answer]
You should be able to pass these variables to your mutation and find them in the fourth argument `info` if that is possible, else you'll have to use something like [`cache.inspectFields`](https://formidable.com/open-source/urql/docs/graphcache/cache-updates/#inspecting-arbitary-entities)

**Reply by edw19** [reply]
thanks cache.inspectFields works for me 😁😁😁😁
